internal/terraform/node_resource_abstract_instance.go

```diff
@@ -1180,6 +1181,10 @@ func (n *NodeAbstractResourceInstance) plan(
 		}
 	}
 
+	// We don't need ephemeral values anymore since the value has been planned
+	ephemeralValuePaths, _ := marks.PathsWithMark(unmarkedPaths, marks.Ephemeral)
+	plannedNewVal = ephemeral.RemoveEphemeralValues(plannedNewVal)
```

Comment:
Provider cannot send back ephemeral values, so we should not just be blindly removing the values here. If they did somehow exist, that would be an error.